.azure/pipelines/azure-pipelines-mirror-within-azdo.yml

```
@@ -4,58 +4,66 @@ trigger:
   branches:
     include:
     - internal/release/6.0
+    
+parameters:
+# Run the pipeline manually (usually disallowed)
+- name: manualRun
+  default: false
+  displayName: Are you sure you want to run this pipeline manually?
+  type: boolean
 
 variables:
   - group: Mirror-Credentials
 
 # Merges code from one AzDO branch into another
 jobs:
-- template: /eng/common/templates/jobs/jobs.yml
-  parameters:
-    enableTelemetry: true
-    helixRepo: dotnet/arcade
-    jobs:
-    - job: Merge_Azure_DevOps_Branches
-      enableSBOM: false
-      condition: and(contains(variables['Build.SourceBranch'], 'internal'), eq(variables['Build.Reason'], 'BatchedCI'))
-      pool:
-        name: NetCore1ESPool-Internal
-        demands: ImageOverride -equals Build.Server.Amd64.VS2019
-      variables:
-      - name: WorkingDirectoryName
-        value: repo-dir
-      - name: AzdoRepo
-        value: dotnet-aspnetcore
-      - name: TargetBranchName
-        value: $(Build.SourceBranch)-nonstable
-      - name: BranchToMirror
-        value: $(Build.SourceBranch)
-      steps:
-      - script: |
-          git clone https://dn-bot:$(dn-bot-dnceng-build-rw-code-rw)@dev.azure.com/dnceng/internal/_git/$(AzdoRepo) $(WorkingDirectoryName) --recursive --no-tags --branch $(TargetBranchName)
-        displayName: Clone AzDO repo
-      - script: |
-          git -c user.email="dotnet-bot@microsoft.com" -c user.name="dotnet-bot" merge origin/$(BranchToMirror) -m "Merge in '$(BranchToMirror)' changes" 
-        displayName: Merge head branch to target branch
-        workingDirectory: $(WorkingDirectoryName)
-      - script: |
-          git push origin $(TargetBranchName)
-        displayName: Push changes to Azure DevOps repo
-        workingDirectory: $(WorkingDirectoryName)
-
-      - task: PowerShell@1
-        displayName: Broadcast target, branch, commit in metadata
-        continueOnError: true
-        condition: always()
-        inputs:
-          scriptType: inlineScript
-          arguments: '$(BranchToMirror)'
+- ${{ if or(and(contains(variables['Build.SourceBranch'], 'internal'), eq(variables['Build.Reason'], 'BatchedCI')), eq(parameters.manualRun, 'true')) }}:
```

The main change is moving the old condition from line 20 to here

---

Hmm, I misread this in the previous PR. Should be

```suggestion
- ${{ if or(and(contains(variables['System.TeamProject'], 'internal'), eq(variables['Build.Reason'], 'BatchedCI')), eq(parameters.manualRun, 'true')) }}:
```

I'd actually prefer _not_ to support any public runs of this (should we stupidly set up a public pipeline):

```suggestion
- ${{ if and(contains(variables['Build.SourceBranch'], 'internal'), or(eq(variables['Build.Reason'], 'BatchedCI'), eq(parameters.manualRun, 'true'))) }}:
```

---

I had `System.TeamProject` originally in that PR, but changed it to `Build.SourceBranch` because there is no public pipeline. I don't know why we'd ever set one up. The second condition you posted should cover any situation we'd care about, I'll use that.